src/combinatorics/quiver.lean

```diff
@@ -135,33 +144,34 @@ end path
 
 /-- A quiver is an arborescence when there is a unique path from the default vertex
     to every other vertex. -/
-class arborescence {V} (T : quiver.{v u} V) : Sort (max (u+1) v) :=
+class arborescence (V : Type u) [quiver.{v} V] : Type (max u v) :=
 (root : V)
-(unique_path : Π (b : V), unique (T.path root b))
+(unique_path : Π (b : V), unique (path root b))
 
 /-- The root of an arborescence. -/
-protected def root {V} (T : quiver V) [arborescence T] : V :=
-arborescence.root T
+def root (V : Type u) [quiver.{v} V] [arborescence V] : V :=
```

Comment:
Why not drop the universe in `quiver.{v}`? Is this so `v` and `u` end up in the right order?

Reply:
Typically any time I write something new and run into any universe issue at all, I switch to being completely explicit. Then, at the end, I try to see if removing most the universe annotations makes things easier to read. I hadn't got to that step yet! I'll have a look now. :-)

Reply:
Done.